src/services/accounts/AccountsBalanceInfoService.ts

```diff
@@ -25,9 +25,18 @@ export class AccountsBalanceInfoService extends AbstractService {
 		hash: BlockHash,
 		historicApi: ApiDecoration<'promise'>,
 		address: string,
-		token: string
+		token: string,
+		withDenomination: boolean
 	): Promise<IAccountBalanceInfo> {
 		const { api } = this;
+
+		if (withDenomination && api.registry.chainDecimals.length === 0) {
+			throw new BadRequest(
+				"Invalid use of the query parameter `denominated`. This chain doesn't have a valid chain decimal to denominate a value."
+			);
+		}
+
+		const decimal = api.registry.chainDecimals[0];
```

Comment:
Not important but I wonder whether it's worth a teeny helper function here like:

```js
function inDenomination(value) {
    withDenomination ? this.applyDenomination(value, decimal) : value
}
```

Just so that we don't need to repeat the ternary below?

Reply:
Yea I agree this is cleaner.

Reply:
fixed